File: src/app/models/LoanAccount.js

// Define collection and schema for LoanAccount
export interface LoanAccount {
    loanNumber:
	type : string
    principalAmount:
	type : Money
    outstandingPrincipal:
	type : Money
    interestRate:
	type : Percentage
    originationDate:
	type : Date
    maturityDate:
	type : Date
    paymentDayOfMonth:
	type : number
    currency:
	type : string
    Bank:
	type : Schema.Types.ObjectId
    Branch:
	type : Schema.Types.ObjectId
    Product:
	type : Schema.Types.ObjectId
    Borrowers:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Customer' }]
    RepaymentSchedule:
 	type : [{ type: Schema.Types.ObjectId, ref: 'RepaymentSchedule' }]
    Payments:
 	type : [{ type: Schema.Types.ObjectId, ref: 'LoanPayment' }]
    Collateral:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Collateral' }]
    FeeCharges:
 	type : [{ type: Schema.Types.ObjectId, ref: 'FeeCharge' }]
    LoanType:
 	type : String
    RateType:
 	type : String
    Compounding:
 	type : String
    Status:
 	type : String
#
    collection: 'loanAccounts'
}

File: bankingOnGolang/internal/model/LoanAccount.go

package model

import (
    "time"
)

//==============================================================
// LoanAccount Declaration
//==============================================================
type LoanAccount struct {
    BaseModel
     LoanNumber            string
    PrincipalAmount        Money `gorm:"embedded;embeddedPrefix:loanAccount_principalAmount"`
    OutstandingPrincipal        Money `gorm:"embedded;embeddedPrefix:loanAccount_outstandingPrincipal"`
    InterestRate        Percentage `gorm:"embedded;embeddedPrefix:loanAccount_interestRate"`
    OriginationDate            time.Time
    MaturityDate            time.Time
    PaymentDayOfMonth            int32
    Currency            string
    BankId         *uint
    Bank           *Bank `gorm:"foreignKey:BankId"`
    BranchId         *uint
    Branch           *Branch `gorm:"foreignKey:BranchId"`
    ProductId         *uint
    Product           *BankingProduct `gorm:"foreignKey:ProductId"`
     Borrowers           []Customer `gorm:"foreignKey:BorrowersFromLoanAccountId"`
     RepaymentSchedule           []RepaymentSchedule `gorm:"foreignKey:RepaymentScheduleFromLoanAccountId"`
     Payments           []LoanPayment `gorm:"foreignKey:PaymentsFromLoanAccountId"`
     Collateral           []Collateral `gorm:"foreignKey:CollateralFromLoanAccountId"`
     FeeCharges           []FeeCharge `gorm:"foreignKey:FeeChargesFromLoanAccountId"`
    LoanType            LoanType
    RateType            RateType
    Compounding            InterestCompounding
    Status            LoanStatus

// parent associations as their child

}

